### PR TITLE
feat(core): add support for `U+` - six-digit microsecond format component

### PR DESCRIPTION
**Pull request by bluestreak01**
fixes https://github.com/questdb/questdb/issues/2687

Also fixes ISO week parsing issue, introduced by https://github.com/questdb/questdb/pull/2632

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 64 / 78 (82.05%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/datetime/microtime/TimestampFormatUtils.java|13|21|61.90%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/std/Numbers.java|19|24|79.17%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/MonthTimestampSampler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/date/MicrosOfSecondFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/GenericTimestampFormat.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/YearTimestampSampler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/TimestampFormatCompiler.java|18|18|100.00%|

File: core/src/main/java/io/questdb/std/datetime/microtime/TimestampFormatUtils.java [hunk 1]
```diff
@@ -433,4 +450,14 @@ private static long parseTimestamp(CharSequence value, int lo, int hi) throws Nu
 
     public static void init() {
     }
+
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/std/datetime/microtime/TimestampFormatUtils.java:453]: Testing leftover ?

Comment (bluestreak01) [reply core/src/main/java/io/questdb/std/datetime/microtime/TimestampFormatUtils.java:453]: Oh, yeah, apologies

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 114 / 120 (95.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/std/Numbers.java|19|24|79.17%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/MonthTimestampSampler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/millitime/GenericDateFormat.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/date/MicrosOfSecondFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/PartitionBy.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/millitime/DateFormatCompiler.java|49|49|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/GenericTimestampFormat.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/TimestampFormatUtils.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/YearTimestampSampler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/TimestampFormatCompiler.java|19|19|100.00%|

File: core/src/main/java/io/questdb/std/datetime/microtime/TimestampFormatUtils.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/std/datetime/microtime/TimestampFormatUtils.java]: fixed here: https://github.com/questdb/questdb/pull/2688/commits/5317e6d0f9daddfce22c1d94086534bdaba879a4

**Comment by bluestreak01** [comment]
@amyshwang more docs here. My observations:

`U` flag will parse micros lazily, even though method internally is called "greedy". In that, value `2` will be parsed as `200`. This is for scenarios such as PG, which doesn't export trailing zeroes in timestamps.

This PR introduces flag `U+`, it is used in "patterns" such as "HH:mm:ss.U+". It will lazily parse 6-digit micros. Think of it as `U` on steroids. `U+` will parse `2` as `200000`

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 121 / 127 (95.28%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/std/Numbers.java|19|24|79.17%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/MonthTimestampSampler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/millitime/GenericDateFormat.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/date/MicrosOfSecondFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/PartitionBy.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/millitime/DateFormatCompiler.java|49|49|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/GenericTimestampFormat.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/TimestampFormatUtils.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/YearTimestampSampler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/TimestampFormatCompiler.java|21|21|100.00%|

**Review by ideoma — APPROVED** [review]
good stuff

**Pull request merged**